code/common/datadog.q

```diff
@@ -0,0 +1,36 @@
+// DATADOG CHECKS
+
+//Create datadog namespace
+\d .dg
+
+ddconfigfile:@[value;`ddconfigfile;hsym `$getenv[`KDBAPPCONFIG],"/ddconfig.txt"]
+
+//sets dogstatsd_port to the port defined by ddconfigfile
+$[`ddconfig.txt in key hsym `$getenv[`KDBAPPCONFIG];value each read0 ddconfigfile;dogstatsd_port:8125]
+
+//Functions are set to return 1b or 0b based on the health of the service.
+//Default check returns 1b from each process to indicate process is up and can be queried.
+//These checks run every time crontab/timer is run.
+
+handlers:(`symbol$())!()
+
+is_ok:{[x]
+        f:$[@[{[x].proc.proctype in key .dg.handlers};`;0b];
+                .dg.handlers[.proc.proctype];
+                .dg.default_is_ok];
+        @[f;`;0b]
+        }
+
+default_is_ok:{[x]1b}
+
+//functions to send metrics and events to datadog from TorQ processes
+sendMetric:{[metric_name;metric_value] system"bash -c \"echo  -n '",metric_name,":",(string metric_value),"|g|#shell' > /dev/udp/127.0.0.1/",(string dogstatsd_port),"\"";}
```

Comment:
Was opening a handle to the file/location ever tested as an alternative?

Reply:
alternatively might be possible to use netcat (`nc` command) to make something a bit cleaner?

Reply:
netcat doesn't work well for the sendMetric functionality. Agree about disabling for other platforms though